modules/client/src/client-common/modules/web3.ts

```diff
@@ -5,61 +5,52 @@ import { Contract, ContractInterface } from "@ethersproject/contracts";
 import { Signer } from "@ethersproject/abstract-signer";
 import { GasFeeEstimation } from "../../client-common/interfaces/common";
 import { IClientWeb3Core } from "../interfaces/core";
-
-const daoFactoryAddressMap = new Map<Web3Module, string>();
-const gasFeeEstimationFactorMap = new Map<Web3Module, number>();
-const providersMap = new Map<Web3Module, JsonRpcProvider[]>();
-const providerIdxMap = new Map<Web3Module, number>();
-const signerMap = new Map<Web3Module, Signer>();
+import { NoDaoFactory, NoPluginRepoRegistry, NoDaoRegistry } from "@aragon/sdk-common";
 
 export class Web3Module implements IClientWeb3Core {
   private static readonly PRECISION_FACTOR_BASE = 1000;
+  private providerIdx = -1;
+  private providers: JsonRpcProvider[] = [];
+  private signer?: Signer;
+  private daoFactoryAddress?: string;
+  private daoRegistryAddress?: string;
+  private pluginRepoRegistryAddress?: string;
+  private gasFeeEstimationFactor = 1;
 
   constructor(context: Context) {
```

Comment:
Why are we undoing the whole hardening PR here?

Reply:
because the hardening didn't do anything. everything is still exposed publicly and changeable. So the freeze didn't help.
Yes, you cannot change the functionality of a function, but you can get to the maps.

Reply:
Additionally, the freeze happens during construction. This means 1. I can still modify the prototype as an attacker before the 1. instance is created. 
Secondly as an attacker I just need to make sure, that I get loaded before the sdk and do `Object.freeze = () => {}`. This prevents anything from getting frozen.